module-bluetooth/Bluetooth/interface/profiles/GAP/GAP.cpp

```diff
@@ -243,8 +243,20 @@ namespace bluetooth
         case HCI_EVENT_USER_CONFIRMATION_REQUEST: {
             bd_addr_t addr;
             hci_event_user_confirmation_request_get_bd_addr(packet, addr);
-            hci_connection_t *conn = hci_connection_for_bd_addr_and_type(addr, BD_ADDR_TYPE_ACL);
-            hci_send_cmd(&hci_user_confirmation_request_reply, &conn->address);
+
+            auto code = hci_event_user_passkey_notification_get_numeric_value(packet);
+            auto it   = devices().find(addr);
+            if (it == devices().end()) {
+                gap_remote_name_request(addr, PAGE_SCAN_MODE_STANDARD, 0);
```

Comment:
Just a noob bt question. Can this method not get the name of device with given addr? What then?